### PR TITLE
Adding a test to make sure our generated code does not cause collisions

### PR DESCRIPTION
**Pull request by tbombach**
This test would have failed before `usings` directives were removed from generated code. The `HttpOperationResponse` model is a bit silly, but `Action` is a better example. The user has to avoid this model name (for all of the generated languages), simply because it's defined in `System` in C#.

Once we add some more processing to remove the fully qualified names from generated code as much as possible, this test should still pass (since it checks errors and warnings).

### CODE REVIEW

File: src/generator/AutoRest.CSharp.Unit.Tests/NameCollision.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+                var result = await Compile(fileSystem);
+
+                // filter the warnings
+                var warnings = result.Messages.Where(
+                    each => each.Severity == DiagnosticSeverity.Warning
+                            && !SuppressWarnings.Contains(each.Id)).ToArray();
+
+                // use this to dump the files to disk for examination 
+                fileSystem.SaveFilesToTemp("name_collision");
```
Comment (John-Hart) [inline src/generator/AutoRest.CSharp.Unit.Tests/NameCollision.cs]: do you want to save this file every time the test is run?

Comment (tbombach) [inline src/generator/AutoRest.CSharp.Unit.Tests/NameCollision.cs]: Nope, I don't. Thanks for the catch!

**Comment by tbombach** [comment]
@azuresdkci retest this please!

**Comment by fearthecowboy** [comment]
LGTM :+1:

**Comment by tbombach** [comment]
@azuresdkci retest this please!

**Pull request merged**